x-pack/plugins/actions/server/builtin_action_types/xmatters.ts

```diff
@@ -0,0 +1,326 @@
+        validate: curry(validateActionTypeConfig)(configurationUtilities),
+      }),
+      secrets: schema.object(secretSchemaProps, {
+        validate: curry(validateActionTypeSecrets)(configurationUtilities),
+      }),
+      params: ParamsSchema,
+      connector: validateConnector,
+    },
+    executor: curry(executor)({ logger, configurationUtilities }),
+  };
+}
+
+function validateActionTypeConfig(
+  configurationUtilities: ActionsConfigurationUtilities,
+  configObject: ActionTypeConfigType
+): string | undefined {
+  const configuredUrl = configObject.configUrl;
+  const usesBasic = configObject.usesBasic;
+  if (!usesBasic) return;
+  try {
+    if (configuredUrl) {
+      new URL(configuredUrl);
+    }
+  } catch (err) {
+    return i18n.translate('xpack.actions.builtin.xmatters.xmattersConfigurationErrorNoHostname', {
+      defaultMessage: 'Error configuring xMatters action: unable to parse url: {err}',
+      values: {
+        err,
+      },
+    });
+  }
+
+  try {
+    if (configuredUrl) {
+      configurationUtilities.ensureUriAllowed(configuredUrl);
+    }
+  } catch (allowListError) {
+    return i18n.translate('xpack.actions.builtin.xmatters.xmattersConfigurationError', {
+      defaultMessage: 'Error configuring xMatters action: {message}',
+      values: {
+        message: allowListError.message,
+      },
+    });
+  }
+}
+
+function validateConnector(
+  config: ActionTypeConfigType,
+  secrets: ActionTypeSecretsType
+): string | null {
+  const { user, password, secretsUrl } = secrets;
+  const { usesBasic, configUrl } = config;
+
+  if (usesBasic) {
+    if (secretsUrl) {
+      return i18n.translate('xpack.actions.builtin.xmatters.shouldNotHaveSecretsUrl', {
+        defaultMessage: 'secretsUrl should not be provided when usesBasic is true',
+      });
+    }
+    if (user == null) {
+      return i18n.translate('xpack.actions.builtin.xmatters.missingUser', {
+        defaultMessage: 'Provide valid Username',
+      });
+    }
+    if (password == null) {
+      return i18n.translate('xpack.actions.builtin.xmatters.missingPassword', {
+        defaultMessage: 'Provide valid Password',
+      });
+    }
+    if (configUrl == null) {
+      return i18n.translate('xpack.actions.builtin.xmatters.missingConfigUrl', {
+        defaultMessage: 'Provide valid URL',
+      });
+    }
+  } else {
+    if (user || password) {
+      return i18n.translate('xpack.actions.builtin.xmatters.shouldNotHaveUsernamePassword', {
+        defaultMessage: 'Username and password should not be provided when usesBasic is false',
+      });
+    }
+    if (configUrl) {
+      return i18n.translate('xpack.actions.builtin.xmatters.shouldNotHaveConfigUrl', {
+        defaultMessage: 'configUrl should not be provided when usesBasic is false',
+      });
+    }
+    if (secretsUrl == null) {
+      return i18n.translate('xpack.actions.builtin.xmatters.missingSecretsUrl', {
+        defaultMessage: 'Provide valid URL with API Key',
```

Comment:
same as ^^^ - let's be more specific on the property name, which is `secretsUrl` in this case.